pkg/osvscanner/customgitignore/dir.go

```diff
@@ -0,0 +1,86 @@
+// based on https://github.com/go-git/go-git/blob/v5.7.0/plumbing/format/gitignore/dir.go
+// but modified so that it skips ignored directories while traversing for gitignore files
+
+package customgitignore
+
+import (
+	"bufio"
+	"os"
+	"strings"
+
+	"github.com/go-git/go-billy/v5"
+	"github.com/go-git/go-git/v5/plumbing/format/gitignore"
+)
+
+const (
+	commentPrefix   = "#"
+	gitDir          = ".git"
+	gitignoreFile   = ".gitignore"
+	infoExcludeFile = gitDir + "/info/exclude"
+)
+
+// readIgnoreFile reads a specific git ignore file.
+func readIgnoreFile(fs billy.Filesystem, path []string, ignoreFile string) (ps []gitignore.Pattern, err error) {
```

Comment:
I'm guessing the path as an slice/array is the case upstream? 

It seems bit weird that all this seems to do is force us to do a split when calling `readIgnoreFile` and `readPattern`, and immediately gets joined back together in these two functions.

Reply:
Yes. The requirementment comes from upstream. I'm not a fan of thinking about path's this way, but I assume this is related to the fact we're hooking into the internals of a library, and processing dirs this way helps speed/algorthim reasons.

My guess is it makes it easier to work out where you were with respect to the .gitignore that created a pattern.
So a pattern from `REPO_ROOT/dir_a/.gitignore` would only apply to things matching `REPO_ROOT/dir_a/**`.
With this style of `path`, you can efficiently traverse down the tree appending to `path` in every dir.

eg, here's what I got from the tests (note the `domain` field.)

```
&gitignore.pattern{domain:[]string{".", "dir_a", "dir_b"}, pattern:[]string{"DIR_B_GITIGNORE"}, inclusion:false, dirOnly:false, isGlob:false} 

```